content/en/tracing/setup_overview/custom_instrumentation/dotnet.md

```diff
@@ -101,9 +100,61 @@ catch(Exception e)
+
+If modifying application code is not possible, use the environment variable `DD_TRACE_METHODS` to detail these methods.
+
+If you have existing `[Trace]` or similar attributes, or prefer to use attributes to complete any incomplete traces within Datadog, use Trace Annotations.
+
+### Instrument methods via configuration
+
+<div class="alert alert-warning">
+  <strong>Note:</strong> This requires enabling automatic instrumentation for your application.
+</div>
+
+Using the `DD_TRACE_METHOD` environment variable, you can get visibility into unsupported frameworks without changing application code. For full details on the input format, see the [.NET Framework setup instructions][8] or the [.NET Core setup instructions][9]. For the following example, assume that the desired method to instrument is named `SaveSession` and the method is defined on the `Store.Managers.SessionManager` type:
+
+```ini
+DD_TRACE_METHODS=Store.Managers.SessionManager[SaveSession]
+```
+
+The resulting span has an `operationName` set to `trace.annotation` and `resourceName` set to `SaveSession`. If you would like to customize the span's attributes and you have the ability to modify the source code, you can [instrument methods via attributes][10] instead.
+
+### Instrument methods via attributes
+
+<div class="alert alert-warning">
+  <strong>Note:</strong> This requires adding the `Datadog.Trace.Annotations` NuGet package and enabling automatic instrumentation for your application.
+</div>
+
+Add `[Trace]` to methods to have them be traced when running with automatic instrumentation. If automatic instrumentation is not enabled, this attribute has no effect on your application.
+
+`[Trace]` attributes have the default operation name `trace.annotation` and resource name of the traced method. These can be set as named arguments of the `[Trace]` attribute to better reflect what is being instrumented.  These are the only possible arguments that can be set for the `[Trace]` attribute.
+
+```csharp
+using Datadog.Trace.Annotations;
+
+namespace Store.Managers
+{
+    public class SessionManager
+    {
+        [Trace(OperationName = "database.persist", ResourceName = "SessionManager.SaveSession")]
+        public static void SaveSession()
+        {
+            // your method implementation here
+        }
+    }
+}
+```
+
 ### Manually creating a new span
 
-Customize your observability by programmatically creating spans around any block of code. Spans created in this manner integrate with other tracing mechanisms automatically. In other words, if a trace has already started, the manual span has its caller as its parent span. Similarly, any traced methods called from the wrapped block of code have the manual span as its parent.
+<div class="alert alert-warning">
+  <strong>Note:</strong> This requires adding the `Datadog.Trace` NuGet package to your application.
```

Comment:
```suggestion
  <strong>Note:</strong> Add the `Datadog.Trace` NuGet package to your application before manually creating a new span.
```

Or use other phrasing consistent with the other alerts, but avoid the unclear "This" pronoun.